runtime/runtime/src/system.rs

```diff
@@ -71,8 +71,12 @@ pub fn staking(
                 .map_err(|err| err.to_string())?,
             amount: body.amount,
         });
-        sender.amount -= body.amount;
-        sender.staked += body.amount;
+        if sender.staked >= body.amount {
```

Comment:
We can not just return stake right away.
We must record that stake must be returned and then return at the end of epoch + 2.
I think we need additional information in ValidatorManager, that tracks "kicked out" stakers and then NightshadeRuntime apply_transactions at the "start of new epoch" block, can fetch it and return it. That also will be the place to calculate reward and update validators balances.

One thing to consider for reward - do we want to apply reward to balance or to the stake? E.g. do people want to reinvest their reward right away into stake.